<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DlgSettings</class>
 <widget class="QDialog" name="DlgSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>533</width>
    <height>481</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resource/resource.qrc">
    <normaloff>:/16x16_ico_png/settings.png</normaloff>:/16x16_ico_png/settings.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabEditor">
      <attribute name="title">
       <string>Settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>General settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Application style</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QCheckBox" name="chkSaveAutomatically">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Save changes before play</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1" colspan="2">
             <widget class="QComboBox" name="cmbWinOrTab">
              <item>
               <property name="text">
                <string>Window</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Tab</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="1" colspan="2">
             <widget class="QComboBox" name="cmbStyle"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Window mode</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Window</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_11">
       <item>
        <widget class="QGroupBox" name="groupBox_9">
         <property name="title">
          <string>Window settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="1" column="0">
             <widget class="QLabel" name="label_22">
              <property name="text">
               <string>Default clear color</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <layout class="QVBoxLayout" name="verticalLayout_12">
              <item>
               <widget class="QCheckBox" name="chkShowGrid">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Show grid</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="chkShowOrigin">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Show origin</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="chkShowViewport">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Show viewport</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="chkSnapToGrid">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Snap to grid</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>Default zoom level</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="zoom">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <double>0.100000000000000</double>
              </property>
              <property name="maximum">
               <double>5.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
              <property name="value">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="color_widgets::ColorSelector" name="clearColor">
                <property name="focusPolicy">
                 <enum>Qt::StrongFocus</enum>
                </property>
                <property name="color">
                 <color>
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </property>
                <property name="display_mode" stdset="0">
                 <enum>color_widgets::ColorPreview::SplitAlpha</enum>
                </property>
                <property name="background">
                 <brush brushstyle="Dense3Pattern">
                  <color alpha="255">
                   <red>128</red>
                   <green>128</green>
                   <blue>128</blue>
                  </color>
                 </brush>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="btnResetClearColor">
                <property name="text">
                 <string>Browse ...</string>
                </property>
                <property name="icon">
                 <iconset resource="../resource/resource.qrc">
                  <normaloff>:/16x16_ico_png/reset.png</normaloff>:/16x16_ico_png/reset.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_16">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Default grid</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="cmbGrid">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>34</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabApps">
      <attribute name="title">
       <string>Applications</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Image editor (.png, .jpeg, .bmp etc. files)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Executable</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="edtImageEditorExecutable"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Arguments</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="edtImageEditorArguments">
            <property name="text">
             <string>${file}</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QToolButton" name="btnSelectImageEditor">
            <property name="text">
             <string>Browse ...</string>
            </property>
            <property name="icon">
             <iconset resource="../resource/resource.qrc">
              <normaloff>:/16x16_ico_png/folder.png</normaloff>:/16x16_ico_png/folder.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Shader editor (.glsl files)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Executable</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="edtShaderEditorExecutable"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Arguments</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="edtShaderEditorArguments">
            <property name="text">
             <string>${file}</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QToolButton" name="btnSelectShaderEditor">
            <property name="text">
             <string>Browse ...</string>
            </property>
            <property name="icon">
             <iconset resource="../resource/resource.qrc">
              <normaloff>:/16x16_ico_png/folder.png</normaloff>:/16x16_ico_png/folder.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Script editor (.lua files)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Executable</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="edtScriptEditorExecutable"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Arguments</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="edtScriptEditorArguments">
            <property name="text">
             <string>${file}</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QToolButton" name="btnSelectScriptEditor">
            <property name="text">
             <string>Browse ...</string>
            </property>
            <property name="icon">
             <iconset resource="../resource/resource.qrc">
              <normaloff>:/16x16_ico_png/folder.png</normaloff>:/16x16_ico_png/folder.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>Audio editor (.mp3, .wav, .ogg files)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Executable</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="edtAudioEditorExecutable"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Arguments</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="edtAudioEditorArguments">
            <property name="text">
             <string>${file}</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QToolButton" name="btnSelectAudioEditor">
            <property name="text">
             <string>Browse ...</string>
            </property>
            <property name="icon">
             <iconset resource="../resource/resource.qrc">
              <normaloff>:/16x16_ico_png/folder.png</normaloff>:/16x16_ico_png/folder.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>96</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabTools">
      <attribute name="title">
       <string>Tools</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QGroupBox" name="groupBox_7">
         <property name="title">
          <string>Python executable</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLineEdit" name="edtPythonExecutable">
              <property name="placeholderText">
               <string>Select python executable</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btnSelectPython">
              <property name="text">
               <string>Browse ...</string>
              </property>
              <property name="icon">
               <iconset resource="../resource/resource.qrc">
                <normaloff>:/16x16_ico_png/folder.png</normaloff>:/16x16_ico_png/folder.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Select the Python executable here. It is needed in order to invoke Emscripten Python scripts for packaging content for web and WASM.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_8">
         <property name="title">
          <string>Emscripten path</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLineEdit" name="edtEmscriptenPath">
              <property name="placeholderText">
               <string>Select emsdk path</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="btnSelectEmsdk">
              <property name="text">
               <string>Browse ...</string>
              </property>
              <property name="icon">
               <iconset resource="../resource/resource.qrc">
                <normaloff>:/16x16_ico_png/folder.png</normaloff>:/16x16_ico_png/folder.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Select the Emscripten SDK path here. It is needed in order to invoke Emscripten Python scripts for packaging content for web and WASM.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabEditor">
      <attribute name="title">
       <string>Text Editor</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Text editor</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="2" column="1">
             <widget class="QCheckBox" name="editorShowLineNumbers">
              <property name="text">
               <string>Show line numbers</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="editorTheme"/>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_9">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Font</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QComboBox" name="editorFontSize"/>
            </item>
            <item row="0" column="1">
             <widget class="QFontComboBox" name="editorFont"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_10">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Theme</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QCheckBox" name="editorHightlightCurrentLine">
              <property name="text">
               <string>Highlight current line</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QCheckBox" name="editorHightlightSyntax">
              <property name="text">
               <string>Highlight syntax</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QCheckBox" name="editorInsertSpaces">
              <property name="text">
               <string>Insert spaces on Tab</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnAccept">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>color_widgets::ColorPreview</class>
   <extends>QWidget</extends>
   <header>color_preview.hpp</header>
  </customwidget>
  <customwidget>
   <class>color_widgets::ColorSelector</class>
   <extends>color_widgets::ColorPreview</extends>
   <header>color_selector.hpp</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>cmbStyle</tabstop>
  <tabstop>cmbWinOrTab</tabstop>
  <tabstop>chkSaveAutomatically</tabstop>
  <tabstop>cmbGrid</tabstop>
  <tabstop>clearColor</tabstop>
  <tabstop>btnResetClearColor</tabstop>
  <tabstop>zoom</tabstop>
  <tabstop>chkShowGrid</tabstop>
  <tabstop>chkShowOrigin</tabstop>
  <tabstop>chkShowViewport</tabstop>
  <tabstop>chkSnapToGrid</tabstop>
  <tabstop>edtImageEditorExecutable</tabstop>
  <tabstop>btnSelectImageEditor</tabstop>
  <tabstop>edtImageEditorArguments</tabstop>
  <tabstop>edtShaderEditorExecutable</tabstop>
  <tabstop>btnSelectShaderEditor</tabstop>
  <tabstop>edtShaderEditorArguments</tabstop>
  <tabstop>edtScriptEditorExecutable</tabstop>
  <tabstop>btnSelectScriptEditor</tabstop>
  <tabstop>edtScriptEditorArguments</tabstop>
  <tabstop>edtAudioEditorExecutable</tabstop>
  <tabstop>btnSelectAudioEditor</tabstop>
  <tabstop>edtAudioEditorArguments</tabstop>
  <tabstop>edtPythonExecutable</tabstop>
  <tabstop>btnSelectPython</tabstop>
  <tabstop>edtEmscriptenPath</tabstop>
  <tabstop>btnSelectEmsdk</tabstop>
  <tabstop>editorFont</tabstop>
  <tabstop>editorFontSize</tabstop>
  <tabstop>editorTheme</tabstop>
  <tabstop>editorShowLineNumbers</tabstop>
  <tabstop>editorHightlightCurrentLine</tabstop>
  <tabstop>editorHightlightSyntax</tabstop>
  <tabstop>editorInsertSpaces</tabstop>
  <tabstop>btnAccept</tabstop>
  <tabstop>btnCancel</tabstop>
 </tabstops>
 <resources>
  <include location="../resource/resource.qrc"/>
 </resources>
 <connections/>
</ui>
